scheduler/scheduler.go

```diff
@@ -0,0 +1,50 @@
+package scheduler
+
+import (
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/executor"
+)
+
+/*
+	"No one has ever looked at a cloud scheduler and thought, 'this is everything I need!'".
+
+	Schedulers manage a stream of Formulas and return running Jobs.
+	These Jobs might be scheduled locally or over an enormous cluster of remote machines.
+
+	Because these imply significant complexity, it would be naive to assume that a scheduler interface could be one-size-fits-all.
+	This interface comprises an optimistic starting point that schedulers can follow, but may surpass.
+
+	Schedulers are presumed to know environmental context that a Formula provider may not.
+	Any problems with a transport running on a remote host, for example, is the scheduler's problem.
+
+	REVIEW: our interfaces do not support returning a guid from scheduling, as executors create their guid when starting.
+	It would be cleaner to be able to refer to a scheduled Job without waiting arbitrarily long for the scheduler to start the job.
```

Comment:
I'm all in favor of this: setting up the guid so we can talk about a job before it's been launched is totally correct.  Looks like you already made this change to Executors... should it push out through schedulers too?  I'm not really sure, since we haven't doc'd what exactly the scheduler's responsibility level is -- is it just supposed to handle pushing tasks out to a farm, or is it actually the originator of work requests?  Probably the latter, thinking out loud, which would indicate yeah, let's push the guid assignment one level higher yet.